client/src/lib/data.ts

```diff
@@ -36,19 +38,24 @@ export async function fetchTickets(userId: number): Promise<Event> {
 }
 
 export async function purchaseTickets(
-  userId: number,
   eventId: number,
+  userId: number,
   ticketCount: number
 ): Promise<Event> {
+  const auth = localStorage.getItem(tokenKey);
+  const a = JSON.parse(auth as string);
   const req = {
-    method: 'GET',
+    method: 'POST',
     headers: {
       'Content-Type': 'application/json',
-      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
+      Authorization: `Bearer ${a.token}`,
```

Comment:
It's interesting that your code for getting this is different than when you do it up above. Do those endpoints work? (Or did you fix this in a later PR?)